Allow to skip file upload and provide a file of filenames as source for data files

mobilegenome opened the pull request:
Another time-saver for frequent test-submissions: 

- Added an argument `--no-upload` that skips file upload entirely. Note that data files still have to be supplied to generate MD5 sums. 
- Treat argument `--data` as a file of file list if of type `str`. This makes creating the command line easier for large submissions.

bgruening commented on ena_upload/ena_upload.py:
```diff
@@ -654,7 +659,27 @@ def collect_tables(args):
     return schema_tables
 
+def get_data_file_path(data):
+    '''Return a list of data files provided as command line argument.
```
Can we make this explicit with a new parameter? My hope is that it's easier to document and has less magic going on.
What do you think?

mobilegenome replied:
My thought was that it is easier on the user-side this way, also as I see it unlikely that a single file would be submitted. But I also see your point. Having a new parameter would also require an additional check that either `--data` or `--data-list` is provided. (Is argparse allowing for that out of the box?)

bedroesb replied:
I also like more the idea of bgruening, especially to not exclude the possibility to submit only one single read file. Note that the `--data` parameter is not mandatory and people can just submit samples or other types with our tool without  data.

mobilegenome replied:
OK, agreed. In fact, I overlooked the possibility that `--data` is optional if you don't submit a RUN table. So the proposed `--no-upload`  would allow to provide the full set of metadata incl. RUN but without running the upload.

mobilegenome replied:
Function was removed.

bedroesb reviewed (changes requested):
I add this as a small reminder:

- [x] Updating the documentation

mobilegenome commented:
I reverted the method to read a file of filenames, as this behaviour can be more simply archieved using a subshell .e.g `<(cat my-files.txt)`. README is also updated, thanks for the reminder!

bedroesb commented on README.md:
```diff
@@ -153,6 +154,15 @@ Most files uploaded to the ENA FTP server need to be compressed.
+Thus, if multiple test submission are performed, it is possible to skip the data upload with `--no_upload` in
+subsequent submissions.
+This also allows uploading (large) datasets separately e.g. with [aspera](https://ena-docs.readthedocs.io/en/latest/submit/fileprep/upload.html).
+For the `--no_upload` argument,  data file(s) still with to be provided with `--data` 
```
I think this sentence misses something :)

mobilegenome replied:
Whoops, only had one coffee this morning ;)

bedroesb commented:
@mobilegenome thanks a lot for these improvements! I will also make use of this ;)

bedroesb commented:
I am today going to make some changes in how the taxon ID is generated (I think it makes more sense the user gives the taxon ID and that the tool guesses the scientific name than the opposite, but I will keep both options . ) After that we can bumb a new version, and that will be a big one!

bedroesb commented:
@bgruening Do you think we can merge this?

bgruening reviewed (approved):
Yes, this looks great. Thank @mobilegenome!
